digital_asset_types/src/dapi/common/asset.rs

```diff
@@ -151,6 +151,35 @@ pub fn safe_select<'a>(
         .and_then(|v| v.pop())
 }
 
+#[cfg(feature = "fetch-raw-fields")]
+async fn process_raw_fields(
+    name: Option<String>,
+    symbol: Option<String>,
+    asset_data: &AssetData,
+) -> (Option<String>, Option<String>) {
+    let name_result = if let Some(name) = &name {
+        let raw_name = asset_data.raw_name.clone();
+        match String::from_utf8(raw_name) {
+            Ok(s) => Some(s),
+            Err(_) => Some(name.clone()),
```

Comment:
Under what scenarios would this happen? Do you have an example NFT?

Reply:
It can fail if the utf8 check validation of the string fails. I don't have an example nft